### PR TITLE
Add safeOnly UDA, associated tests. Fix issue with opUnary.

### PR DESCRIPTION
**Pull request by schveiguy**
Kind of helps with #39 as my real goal is to have a safe-only algebraic type. But the main crux of that issue is still present (no way to disable other ops), which might have other use cases.

But this is hugely important for my work. I want a fully-safe algebraic type to use in mysql-native.

I also found  a bug in opUnary (apparently nobody ever used it!).

### CODE REVIEW

File: source/taggedalgebraic/taggedalgebraic.d [hunk 1]
```diff
@@ -781,6 +781,28 @@ private struct DisableOpAttribute {
...
+@safe unittest
+{
+	union Fields
+	{
+		int intval;
+		@safeOnly int *ptr;
+	}
+
+	// only safe operations allowed on pointer field
+	@safe test() {
```
Comment (s-ludwig) [inline source/taggedalgebraic/taggedalgebraic.d]: Is this intentionally not executed? It looks like the code could just directly be in the unit test itself, which would then hopefully fix the overall coverage loss.

Comment (schveiguy) [reply source/taggedalgebraic/taggedalgebraic.d]: Well, not intentionally. It was meant to show in docs that the code can be done in safe mode (not sure if safe unittests show as such). I didn’t even consider that it is why code coverage is failing. Will fix.

**Comment by schveiguy** [comment]
I included the test call. I'm looking at the coverage report, and I'm not sure what I'm doing wrong to have 2 of them fail. Looking at the coverage report, it's highlighting blank lines as not being covered?

**Comment by s-ludwig** [comment]
Okay, looks good now (bummer though that the coverage info is not smarter w.r.t. declarations and CTFE)!

**Pull request merged**